lib/sword/endpoints/endpoint.rb

```diff
@@ -15,6 +15,20 @@ def initialize(collection_slug, depositor_user_id)
         @collection_slug = collection_slug
         @depositor_user_id = depositor_user_id
       end
+
+      def self.get_endpoint(collection_slug,
+                            depositor_user_id)
+        case COLLECTIONS[:slug][collection_slug][:parser]
+        when "academic-commons"
+          Sword::Endpoints::AcademicCommonsEndpoint.new(collection_slug, depositor_user_id)
+        when "proquest"
+          Sword::Endpoints::ProquestEndpoint.new(collection_slug, depositor_user_id)
+        when "eprints"
+          Sword::Endpoints::EprintsEndpoint.new(collection_slug, depositor_user_id)
+        else
+          # raise an exception here
```

Comment:
just flagging this, in case you want to add the exception

Reply:
I'll second the suggestion about adding an exception.  The `# raise an exception here` TODO should be resolved before merging this PR (otherwise it could result in a less helpful null pointer error in `app/helpers/deposits_helper.rb` or `app/controllers/sword_controller.rb`).

Reply:
Plus you'll want to add a test for that `else` case.